Library/Homebrew/utils/lock.sh

```diff
@@ -0,0 +1,40 @@
+# Ref: http://wiki.bash-hackers.org/howto/mutex
+
+HOMEBREW_LOCK_DIR="$HOMEBREW_LIBRARY/lock.d"
+HOMWBREW_LOCK_PID_FILE="$HOMEBREW_LOCK_DIR/PID"
+
+lock() {
```

Comment:
I think that approach could work fine for both but if other maintainers don't agree I don't feel strongly.